File: Modules/Common/Items/SIE.Web.Items/ProductModels/Commands/ProductModelSaveCommand.js
SIE.defineCommand('SIE.Web.Items.ProductModels.Commands.ProductModelSaveCommand', {
    extend: 'SIE.cmd.Save',
    meta: { text: "保存", group: "edit", iconCls: "icon-SaveEntity icon-blue" },
    canExecute: function (view) {
        var result = false;
        result = view.getData().isDirty();
        if (result == false) {
            var pv = view.getChildren();
            var current = view.getCurrent();
            if (current) {
                for (var i = 0; i < pv.length - 1; i++)
                {
                    if (pv[i].getData().isDirty()) {
                        return true;
                    }
                }
            }
        }
        return result;
    },
    execute: function (view, source) {
        var me = this;
        var children = view.getChildren();
        var indata = {};
        var productModel = [];
        var productModelLineCapacity = [];
        var productModelSkill = [];
        var data = view.getData();
        var childrenData1 = children[0].getData();
        var childrenData2 = children[1].getData();
        for (var i = 0; i < data.data.items.length; i++) {
            productModel[i] = data.data.items[i].data;
        }
        for (var i = 0; i < childrenData1.data.items.length; i++) {
            productModelSkill[i] = childrenData1.data.items[i].data;
        }
        for (var i = 0; i < childrenData2.data.items.length; i++) {
            productModelLineCapacity[i] = childrenData2.data.items[i].data;
        }

        indata.Data = Ext.encode({ ProductModelList: productModel, ProductModelLineCapacityList: productModelLineCapacity, ProductModelSkillList: productModelSkill  });
        view.execute({
            data: indata,
            success: function (res) {
                var operationView = view;
                if (view.associateCmd) {
                    operationView = view.associateCmd.view;
                }
                me._viewReload(operationView);
                view.getData().dirty = false;
                view.getData().phantom = false;
                SIE.Msg.showInstantMessage('保存成功'.t());
            }
        });
    }

});

File: Modules/Common/Items/SIE.Web.Items/ProductModels/Commands/ProductModelSaveCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.Items;
using SIE.Web.Command;
using System;
using System.Linq;

namespace SIE.Web.Items.ProductModels.Commands
{
    /// <summary>
    /// 保存
    /// </summary>
    [JsCommand("SIE.Web.Items.ProductModels.Commands.ProductModelSaveCommand")]
    public class ProductModelSaveCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<ProductModelSaveViewArgs>();

            //添加检查，这段校验是解决平台报了两个相同的非空提示，其他校验交给平台
            data.ProductModelLineCapacityList.ForEach(a =>
            {
                if (a.ResourceId == 0)
                {
                    throw new ValidationException("资源不能为空".L10N());
                }
            });
            data.ProductModelSkillList.ForEach(a =>
            {
                if (a.SkillId == 0)
                {
                    throw new ValidationException("技能清单不能为空".L10N());
                }
            });

            RF.Save(data.ProductModelList);
            RF.Save(data.ProductModelLineCapacityList);
            RF.Save(data.ProductModelSkillList);
            return true;
        }
    }

    /// <summary>
    /// 参数
    /// </summary>
    public class ProductModelSaveViewArgs
    {
        /// <summary>
        /// 产品机型
        /// </summary>
        public EntityList<ProductModel> ProductModelList { get; set; }

        /// <summary>
        /// 产线产能
        /// </summary>
        public EntityList<ProductModelLineCapacity> ProductModelLineCapacityList { get; set; }

        /// <summary>
        /// 机型技能
        /// </summary>
        public EntityList<ProductModelSkill> ProductModelSkillList { get; set; }
    }
}
